datalad/core/distributed/push.py

```diff
@@ -113,9 +113,10 @@ class Push(Interface):
             args=("--since",),
             constraints=EnsureStr() | EnsureNone(),
             doc="""specifies commit-ish (tag, shasum, etc.) from which to look for
-            changes to decide whether pushing is necessary.
-            If '^' is given, the last state of the current branch at the sibling
-            is taken as a starting point."""),
+            changes to decide whether pushing of data or recursion into subdatasets
```

Comment:
Perhaps "transferring data" rather than "pushing of data"?  My thought is that "transfer" would require fewer brain cycles from the reader to connect up with annex data given the option names (force's current `{no-,}datatransfer`), though that of course would carry less weight with the `--transfer-data -> --data` rename in gh-4620.

Then again there's probably no way around that the caller needing to realize that "data" in the context of `push` (and probably lots of other spots in datalad) means "annex data", so there's probably not much point in the making this change.